### PR TITLE
Move uninstall code to pkg

### PR DESCRIPTION
**Pull request by phillebaba**
This change moves the uninstall logic from `cmd` to `pkg`. This is done so that it can be shared with the Terraform Provider reducing the need to duplicate code.

### CODE REVIEW

File: pkg/uninstall/uninstall.go [hunk 1]
```diff
@@ -0,0 +1,321 @@
+package uninstall
```
Comment (hiddeco) [inline pkg/uninstall/uninstall.go:1]: License header is also missing here :-)

File: pkg/uninstall/uninstall.go [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+					logger.Failuref("CustomResourceDefinition/%s deletion failed: %s", r.Name, err.Error())
+				} else {
+					logger.Successf("CustomResourceDefinition/%s deleted %s", r.Name, dryRunStr)
+				}
+			}
+		}
+	}
+}
+
+func UninstallNamespace(ctx context.Context, logger log.Logger, kubeClient client.Client, namespace string, dryRun bool) {
```
Comment (hiddeco) [inline pkg/uninstall/uninstall.go:291]: Would `uninstall.Namespace` be better?

**Comment by hiddeco** [comment]
Same suggestions apply for the other functions.

**Comment by phillebaba** [comment]
@hiddeco I agree that removing the `Uninstall` prefix from each function seems reasonable as it is in the package name. Fixed your comments.

**Comment by hiddeco** [comment]
One additional nit (other than that absolutely perfect): given things move into the public API, it would probably be good to document the (behavior) of the functionalities. Both because of consumer ease, and ensuring the behavior stays the same in terms of backwards compatibility.

**Review by hiddeco — APPROVED** [review]

**Pull request merged**